nemo/collections/asr/parts/submodules/rnnt_greedy_decoding.py

```diff
@@ -464,7 +464,7 @@ def _greedy_decode_blank_as_pad(
         if partial_hypotheses is not None:
             raise NotImplementedError("`partial_hypotheses` support is not supported")
 
-        with torch.no_grad():
+        with torch.inference_mode():
```

Comment:
Is there any difference between no_grad and inference_mode?

Reply:
Yeah, its a subtle but annoying source of memory leaks, especially during autoregressive decoding. For reference https://pytorch.org/docs/stable/generated/torch.inference_mode.html will not track either gradients or the graph itself - and the graph wastes GPU memory (especially when run autoregressive where you might mistakenly track the variable of the previous step thus wasting memory per step).